gripper_controllers/include/gripper_controllers/gripper_action_controller_impl.hpp

```diff
@@ -64,6 +64,11 @@ template <const char * HardwareInterface>
 controller_interface::return_type GripperActionController<HardwareInterface>::update(
   const rclcpp::Time & /*time*/, const rclcpp::Duration & /*period*/)
 {
+  if (!rt_active_goal_) {
+    set_hold_position();
+    return controller_interface::return_type::OK;
+  }
```

Comment:
This solution will not work if there is given command from topic and not action. Why not put proper state interface value when starting controller?

Reply:
Do you mean setting a default initial value for the controller? It's potentially dangerous to have the controller start moving servos on startup with a real robot system without an explicit request. When developing on a physical system, you can't assume that the robot will be in the same physical state on startup every time. The joint_trajectory_controller operates in the same way. When you start the ros2 control system, it doesn't start moving the servos to an initial value until a plan is requested.

Reply:
I only tested this with the action interface, not the topic interface. I'll look into how to achieve the same effect for both interfaces.

Reply:
@SyllogismRXS did you have time to make this also working on topic interface?

We should probably get state interface from the gripper on the beginning and set commands to states in `on_activate`.

Reply:
We should for sure have the following:

- in `on_activate` commands should be assigned to states; something like here storing into last_commanded_state_; maybe some variables like this if there is no other ([example](https://github.com/ros-controls/ros2_controllers/blob/afe216fa016150a493a29ae7e452fd5e8eef81ec/admittance_controller/src/admittance_controller.cpp#L372)
- check in update is new goal/command has arrived. The best would be to store the goal/command in a realtime buffer es done in JTC [example](https://github.com/ros-controls/ros2_controllers/blob/76bf6ae70c432565c8767fba14c9adc75028292e/joint_trajectory_controller/src/joint_trajectory_controller.cpp#L1218) and then using this in the update loop [like here](https://github.com/ros-controls/ros2_controllers/blob/76bf6ae70c432565c8767fba14c9adc75028292e/joint_trajectory_controller/src/joint_trajectory_controller.cpp#L136)
This could be done exactly like this using pointers. If a command is sent, then one could reset it to `nullptr`.

Does this help a bit?